### ISSUE TITLE
Support `query` variable

### ISSUE THREAD

**Issue by dinoboff** [goldibex/targaryen#142]
The rule parser and evaluator needs to allow it and APIs needs a way to set it.

It's not yet part of the security variables reference but there's a blog post and a guide entry:

- https://firebase.google.com/docs/database/security/securing-data#query_based_rules
- https://firebase.googleblog.com/2018/01/introducing-query-based-security-rules.html

**Comment by dinoboff** [comment]
`query.orderByChild` default value should be `null` according the the guide, but `query.orderByChild != null` is currently an invalid statement. And although, the other `orderBy` attributes default value should be false, `query.orderByValue != null` is valid.

I opened an issue with Firebase to check if it's a bug before accommodating for the new inferring requirement.

**Comment by dinoboff** [comment]
The issue is now fix, and `query.orderByKey`, `query.orderByValue` and `query.orderByPriority`  still support comparison to `null` which match `targaryen` behaviour with auth property access.

**Comment by dinoboff** [comment]
With #144, you can set `query` in `chai` with:
```js
targaryen.setFirebaseData(null);
targaryen.setFirebaseRules({rules: {
  '.read': 'query.orderByChild == "owner" && query.equalTo == auth.uid'
}});

expect({uid: 'bob'}).can.readWith({query: {
  orderByChild: 'owner',
  equalTo: 'bob'
}}).path('/');
```

With `jasmine`:
```js
targaryen.setFirebaseData(null);
targaryen.setFirebaseRules({rules: {
  '.read': 'query.orderByChild == "owner" && query.equalTo == auth.uid'
}});

expect({uid: 'bob'}).canRead('/', {query: {
  orderByChild: 'owner',
  equalTo: 'bob'
}});
```

I will release in 3.1.0-beta.1 before committing to those assertion methods.

**Comment by dinoboff** [comment]
You can try the new matcher/options with ` npm install targaryen@beta`

**Comment by dinoboff** [comment]
Published with `v3.1.0`.
